presentation/src/main/java/me/androidbox/presentation/event/viewmodel/EventViewModel.kt

```diff
@@ -197,6 +197,16 @@ class EventViewModel @Inject constructor(
             is EventScreenEvent.CheckVisitorExists -> {
                 verifyVisitorEmail(eventScreenEvent.visitorEmail)
             }
+            is EventScreenEvent.OnShowDeleteEventAlertDialog -> {
+                _eventScreenState.update { eventScreenState ->
+                    eventScreenState.copy(
+                        shouldShowDeleteAlertDialog = eventScreenEvent.shouldShowDeleteAlertDialog
+                    )
+                }
+            }
+            is EventScreenEvent.OnDeleteEvent -> {
+                deleteEvent(eventScreenEvent.eventId)
```

Comment:
I recommend wrapping this code in an if statement to see whether the user is event creator or not

Reply:
@MKiperszmid @philipplackner I do that check here so that can only happen when that button is shown. And it can only show if the user is a creator. Would that be ok?

```
AgendaAction(
                        agendaActionType = if(eventScreenState.isUserEventCreator) { AgendaActionType.DELETE_EVENT } else { AgendaActionType.JOIN_EVENT },
                        onActionClicked = { agendaActionType ->
                            when(agendaActionType) {
                                AgendaActionType.DELETE_EVENT -> {
                                    eventScreenEvent(EventScreenEvent.OnShowDeleteEventAlertDialog(shouldShowDeleteAlertDialog = true))
                                }
                                AgendaActionType.JOIN_EVENT -> {

                                }
                                AgendaActionType.LEAVE_EVENT -> {

                                }
                                else -> Unit
                            }

                        }
                    )

```

Reply:
Ah yea, totally fine